gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -827,19 +781,62 @@ func (vm *VMKeeper) QueryFile(ctx sdk.Context, filepath string) (res string, err
 		if memFile == nil {
 			return "", fmt.Errorf("file %q is not available", filepath) // TODO: XSS protection
 		}
+
 		return memFile.Body, nil
-	} else {
-		memPkg := store.GetMemPackage(dirpath)
-		if memPkg == nil {
-			return "", fmt.Errorf("package %q is not available", dirpath) // TODO: XSS protection
+	}
+
+	memPkg := store.GetMemPackage(dirpath)
+	if memPkg == nil {
+		return "", fmt.Errorf("package %q is not available", dirpath) // TODO: XSS protection
+	}
+
+	for i, memfile := range memPkg.Files {
+		if i > 0 {
+			res += "\n"
+		}
+		res += memfile.Name
+	}
+
+	return res, nil
+}
+
+func stringifyResultValues(m *gno.Machine, format Format, values []gno.TypedValue) string {
+	switch format {
+	case FormatString:
+		if len(values) != 1 {
+			panic(fmt.Errorf("expected 1 string result, got %d", len(values)))
+		}
+
+		tv := values[0]
+		switch bt := gno.BaseOf(tv.T).(type) {
+		case gno.PrimitiveType:
+			if bt.Kind() == gno.StringKind {
+				return tv.GetString()
+			}
+		case *gno.PointerType:
+			if tv.IsStringer() {
+				res := m.Eval(gno.Call(gno.Sel(&gno.ConstExpr{TypedValue: tv}, "String")))
+				return strconv.Quote(res[0].GetString())
+			}
 		}
-		for i, memfile := range memPkg.Files {
+
+		panic(fmt.Errorf("expected 1 `string` or `Stringer` result, got %v", tv.T.Kind()))
+
+	case FormatJSON:
+		return JSONPrimitiveValues(m, values)
+	case FormatDefault, "":
+		var res strings.Builder
+
+		for i, v := range values {
 			if i > 0 {
-				res += "\n"
+				res.WriteRune('\n')
 			}
-			res += memfile.Name
+			res.WriteString(v.String())
 		}
-		return res, nil
+
```

Comment:
can't you add the two `\n` here and remove the if condition here: https://github.com/gnolang/gno/pull/2949/files#diff-3663e30f0c0e01d8d47bb7974d605b4cf12d1a244970c6a9bbba55fb0be6acedR503 ?